src/angular/17-order-history-component/list.component.ts

```diff
@@ -1,36 +1,33 @@
-import { Component, OnInit, Input } from '@angular/core';
-import { Order, Item, OrderService } from '../order.service';
-
+import { Component, Input, OnInit } from '@angular/core';
+import { Item, Order, OrderService } from '../order.service';
 
 @Component({
   selector: 'pmo-list',
   templateUrl: './list.component.html',
-  styleUrls: ['./list.component.less']
+  styleUrls: ['./list.component.less'],
 })
 export class ListComponent implements OnInit {
-  @Input() orders: Order[];
-  @Input() listTitle: string;
-  @Input() status: string;
-  @Input() statusTitle: string;
-  @Input() action: string;
-  @Input() actionTitle: string;
-  @Input() emptyMessage: string;
+  @Input() orders?: Order[];
+  @Input() listTitle?: string;
+  @Input() status?: string;
+  @Input() statusTitle?: string;
+  @Input() action?: string;
+  @Input() actionTitle?: string;
+  @Input() emptyMessage?: string;
 
-  constructor(private orderService: OrderService) { }
+  constructor(private orderService: OrderService) {}
 
-  ngOnInit() {}
+  ngOnInit(): void {}
 
-  markAs(order: Order, action: string) {
-    this.orderService.updateOrder(order, action).subscribe(() => {
-    });
+  markAs(order: Order, action: string): void {
+    this.orderService.updateOrder(order, action).subscribe();
   }
 
-  delete(id:string) {
-    this.orderService.deleteOrder(id).subscribe(() => {
-    })
+  delete(id: string): void {
+    this.orderService.deleteOrder(id).subscribe();
```

Comment:
should prob throw a takeUntil here for now

Reply:
this is an http request so it is not needed (it always completes)